gslib/commands/test.py

```diff
@@ -292,7 +292,7 @@ def CreateTestProcesses(parallel_tests, test_index, process_list, process_done,
     Index of last created test.
   """
   orig_test_index = test_index
-  executable_prefix = [sys.executable] if sys.executable and IS_WINDOWS else []
+  executable_prefix = [sys.executable] if sys.executable else []
```

Comment:
I think this was originally added with the intention of correctly running from a Python archive file, which bundles the interpreter and the code together. Context, if you're interested: https://github.com/google/subpar.

Can you add a bit to this that checks if we're running in an archive, and if so, doesn't add `sys.executable`? Something like this should do:

```python
  def InvokedFromParFile():
    loader = globals().get('__loader__', None)
    if not loader:
      return False
    return 'zipimport' in loader.__class__.__module__

  executable_prefix = []
  if not InvokedFromParFile():
    executable_prefix.append(sys.executable)

  orig_test_index = test_index
  # ...
```

Reply:
Code in second commit address the changes requested to omit executable prefix when run from python archive files